tools/benchcomp/configs/perf-regression.yaml

```diff
@@ -0,0 +1,41 @@
+# Copyright Kani Contributors
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+#
+# Run the Kani perf suite twice, erroring out on regression. This config
+# file is primarily intended to be used in CI, because it assumes that
+# there are two Kani checkouts in the 'base' and 'pr' directories;
+# benchcomp compares the performance of these two checkouts.
+
+variants:
+  kani_pr:
+    config:
+      directory: pr
+      command_line: PATH=$(realpath pr/scripts):$PATH scripts/kani-perf.sh
+      env:
+        RUST_TEST_THREADS: "1"
+  kani_base:
+    config:
+      directory: base
+      command_line: PATH=$(realpath base/scripts):$PATH scripts/kani-perf.sh
+      env:
+        RUST_TEST_THREADS: "1"
+
+run:
+  suites:
+    kani_perf:
+      parser:
+        module: kani_perf
+      variants: [kani_base, kani_pr]
+
+visualize:
+  - type: error_on_regression
+    variant_pairs: [[kani_base, kani_pr]]
+    checks:
+      - metric: success
+        test: "lambda old, new: False if not old else not new"
+      - metric: solver_runtime
+        test: "lambda old, new: new/old>3 if new < 1 else new/old>2 if new<30 else new/old>1.1"
+      - metric: symex_runtime
+        test: "lambda old, new: new/old>3 if new < 1 else new/old>2 if new<30 else new/old>1.1"
+      - metric: verification_time
+        test: "lambda old, new: new/old>3 if new < 1 else new/old>2 if new<30 else new/old>1.1"
```

Comment:
It seems that symex and solver time produce more stable numbers. So can we please not go ahead with this without first making sure that we are using a metric exhibiting very low noise?

Reply:
I just tried removing Verification Time, and instead checked for solver and symex time >5%. [But one benchmark failed](https://github.com/model-checking/kani/actions/runs/4701956238/jobs/8338630914?pr=2370) ('vec/string/main' regressed on metric 'symex_runtime' (1.5912 -> 1.69417))... so I'm really not convinced that the tool itself is the problem here, the noise seems somewhat intrinsic to the CI machine.

I would like to push back a bit by noting what this PR is for: we're trying to catch changes to Kani (or upgrades to CBMC or the solver) that cause typical benchmarks to regress. I'm not sure that the noise itself is a problem if we can ignore it. If runs of identical Kani are persistently varying by this much, then I think either setting the threshold higher or ignoring fast benchmarks would still allow us to catch real regressions. (I prefer the second approach)